language/tools/test_generation/src/bytecode_generator.rs

```diff
@@ -363,10 +370,10 @@ impl BytecodeGenerator {
             // Add candidates should not be empty unless the list of bytecode instructions is
             // changed
             if add_candidates.is_empty() {
-                panic!("Could not find valid candidate");
+                return Err("Could not find valid add candidate".to_string());
             }
             next_instruction_index = self.rng.gen_range(0, add_candidates.len());
-            add_candidates[next_instruction_index].1.clone()
+            Ok(add_candidates[next_instruction_index].1.clone())
```

Comment:
At this point I find myself wondering why you don't map out the stack effects from add_candidates. I also find it strange that you need to call clone again here.

Reply:
Good point, I've changed it to map those out. Unfortunately that clone can't be avoided.